avro2py/codegen.py

```diff
@@ -363,10 +362,16 @@ def bridge_namespaces(from_namespace: List[str], to_namespace: List[str], target
         remaining_to_components = [c for c in paired_remaining_to_components if c is not None]
 
         if remaining_from_components:
+            level = len(remaining_from_components)
+            for namespace_name in self.namespaces.keys():
+                namespace_parts = namespace_name.split('.')
+                if namespace_parts[0:len(from_namespace)] == from_namespace and len(namespace_parts) > len(from_namespace):
+                    level += 1
+                    break
```

Comment:
I think that rather than doing:
```python
for x in xs:
    if cond(x):
        do_something_not_requiring_x()
        break
```
this would be more clearly expressed if we did:
```python
if any(cond(x) for x in xs):
    do_something_not_requiring_x()
```
, meaning, in this case,
```python
if any(is_child(parent=from_namespace, target=namespace) for namespace in self.namespaces):
    level = len(remaining_from_components) + 1
else:
    level = len(remaining_from_components)
```
(or, if you implement my other comment, 
```python
if from_namespace in self.namespaces_with_children:
    level = len(remaining_from_components) + 1
else:
    level = len(remaining_from_components)
```

Reply:
this is closer to what happens now (although not exactly).